### PR TITLE
Feature/full product

### PR DESCRIPTION
**Pull request by ShumovskaIryna**
Implementation OPEN&CLOSE FullProduct view on click.
Rename Item -> Product (as component's naming in backend)

### CODE REVIEW

File: client/src/components/FullProduct/FullProduct.js [hunk 1]
```diff
@@ -1,38 +1,73 @@
...
+import { useQuery, gql } from '@apollo/client';
+
 import MiniGallery from './MiniGallery';
 import ValueAttributes from './ValueAttributes';
 
-export class FullProduct extends Component {
-  _handleKeyUp = (event) => {
-    if (event.key === 'Escape') {
-      return this.props.onShowProduct(this.props.item);
+const GET_PRODUCT_BY_ID = gql`
```
Comment (vladSkrygun) [inline client/src/components/FullProduct/FullProduct.js]: Lets move this into another file and all other gql string as well

File: client/src/components/FullProduct/FullProduct.js [hunk 2]
```diff
@@ -1,38 +1,73 @@
...
+        amount
+        currency {
+          label
+          symbol
+        }
+      }
+      brand
     }
-  };
-  componentDidMount() {
```
Comment (vladSkrygun) [inline client/src/components/FullProduct/FullProduct.js:11]: Wdyt to leave this listeners?

File: client/src/components/FullProduct/FullProduct.js [hunk 3]
```diff
@@ -1,38 +1,73 @@
...
+    error: errorProduct,
+    data: dataProduct,
+    loading: loadingProduct,
+  } = useQuery(GET_PRODUCT_BY_ID, {
+    variables: {id: props.productId}
+  });
+  console.log(errorProduct)
+  return (
+    <> 
+    {loadingProduct ? <div> Loading...</div>
```
Comment (vladSkrygun) [inline client/src/components/FullProduct/FullProduct.js:21]: maybe we should also add error handling?

File: client/src/components/Products.jsx [hunk 4]
```diff
@@ -45,7 +45,7 @@ const GET_CATEGORIES = gql`
 `;
 
 const Products = () => {
-  const [_, setCategoryInput] = useState('all');
+  const [, setCategoryInput] = useState('all');
```
Comment (vladSkrygun) [inline client/src/components/Products.jsx:8]: Wdyt about creating folder with all constants?

File: client/src/components/Products.jsx [hunk 5]
```diff
@@ -55,31 +55,28 @@ const Products = () => {
 
   console.log('Data', { error, loading, data, errorCategories, loadingCategories });
```
Comment (vladSkrygun) [inline client/src/components/Products.jsx]: please remove it

File: client/src/components/Products.jsx [hunk 6]
```diff
@@ -55,31 +55,28 @@ const Products = () => {
...
 
-  const Loading = () => {
```
Comment (vladSkrygun) [inline client/src/components/Products.jsx:58]: It should be another component

File: client/src/components/Products.jsx [hunk 7]
```diff
@@ -55,31 +55,28 @@ const Products = () => {
...
-          <div className="col-12 mb-5">{data ? <ShowProducts /> : <Loading />}</div>
+          <div className="col-12 mb-5">
+            {data ? (
+              <>
+                <Categories
+                  allCategories={dataCategories?.categories}
+                  setCategory={setCategoryInput}
+                  refetchProducts={refetchProducts}
+                />
+                <div className="container">
```
Comment (vladSkrygun) [inline client/src/components/Products.jsx:30]: Lets move it to another component

**Review by vladSkrygun — APPROVED** [review]
Looks awesome! Well done. I have left a couple comments related to components architecture and in general to code. Let's fix them. It can be done in context of this pr or you can merge this one and open another pr, as you wish.

File: client/src/components/FullProduct/FullProduct.js (hunk 1 above)
Comment (ShumovskaIryna) [reply client/src/components/FullProduct/FullProduct.js]: Sounds great! Thanks!
Done ✅👌

File: client/src/components/FullProduct/FullProduct.js (hunk 2 above)
Comment (ShumovskaIryna) [reply client/src/components/FullProduct/FullProduct.js:11]: Do you mean to CLOSE the FullProduct window by pressing the Escape key?

File: client/src/components/Products.jsx (hunk 4 above)
Comment (ShumovskaIryna) [reply client/src/components/Products.jsx:8]: But I can't do this. React Hooks "useState" & "useQuery" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.

File: client/src/components/FullProduct/FullProduct.js (hunk 3 above)
Comment (ShumovskaIryna) [reply client/src/components/FullProduct/FullProduct.js:21]: Sounds great! Thanks!
Done ✅👌

**Comment by ShumovskaIryna** [comment]
Implementation OPEN&CLOSE FullProduct view on click.
Rename Item -> Product (as component's naming in backend)
create constants folder for queries
error handler for FullProduct page

**Pull request merged**